jax/_src/interpreters/mlir.py

```diff
@@ -1671,29 +1658,6 @@ def wrap_with_memory_kind(
   return op.result
 
 
-def replicate_trailing_dims(ctx, val: ir.Value, aval) -> ir.Value:
```

Comment:
You can't remove this function and it's usage above. This maintains a very important property of extended dtypes that aren't user sharded.

Please restore this back.

Reply:
Done.